src/components/datalist-item/__tests__/DatalistItem.test.tsx

```diff
@@ -1,66 +1,44 @@
 import React from 'react';
 import sinon from 'sinon';
-
+import { shallow } from 'enzyme';
 import DatalistItem from '..';
 
 const sandbox = sinon.sandbox.create();
-
 describe('components/datalist-item/DatalistItem', () => {
     afterEach(() => {
         sandbox.verifyAndRestore();
     });
-
     describe('render()', () => {
         test('should render default component', () => {
             const child = 'Test';
             const wrapper = shallow(<DatalistItem>{child}</DatalistItem>);
-
             expect(wrapper.is('li')).toBe(true);
             expect(wrapper.hasClass('datalist-item')).toBe(true);
-            expect(wrapper.prop('id')).toBeTruthy();
+            expect(wrapper.prop('id')).toMatch(new RegExp('^datalistitem?'));
+            expect(wrapper.prop('role')).toEqual('option');
             expect(wrapper.text()).toEqual(child);
         });
-
         test('should render the correct classes when className and isActive are specified', () => {
             const className = 'test';
             const wrapper = shallow(
                 <DatalistItem className={className} isActive>
                     Test
                 </DatalistItem>,
             );
-
             expect(wrapper.hasClass('datalist-item')).toBe(true);
             expect(wrapper.hasClass('is-active')).toBe(true);
             expect(wrapper.hasClass(className)).toBe(true);
         });
-
         test('should render the correct aria-selected according to isSelected prop', () => {
             const isSelectedValue = true;
             const wrapper = shallow(<DatalistItem isSelected={isSelectedValue}>test</DatalistItem>);
-
             expect(wrapper.prop('aria-selected')).toBe(isSelectedValue);
         });
-
         test('should render custom attributes when specified', () => {
             const wrapper = shallow(<DatalistItem data-resin-target="test">Test</DatalistItem>);
-
             expect(wrapper.prop('data-resin-target')).toEqual('test');
         });
-
-        test('should not override id or role on li when props are specified', () => {
-            const id = 'test';
-            const role = 'listitem';
-            const wrapper = shallow(
-                <DatalistItem id={id} role={role}>
-                    Test
-                </DatalistItem>,
-            );
-
-            expect(wrapper.prop('id')).not.toEqual(id);
-            expect(wrapper.prop('role')).toEqual('option');
-        });
```

Comment:
Are we sure we don't need this?

I think it's testing the order of the props:
`<li {...itemProps} className={classes} id={this.id} role="option" aria-selected={isSelected}>`
so that `itemProps` is not placed after `id={this.id} role="option"`.

Reply:
TS should prevent a dev from setting fields that arent in the props interface, also they somehow get into itemProps, they get overwritten since its being set after the spread.

Reply:
Also when I tried to leave it in there, I was getting TS errors that I was trying to set fields that werent in the interface which was why I had to remove them and decided to combine the assertions with the render test at the top